lib/screens/create_region_event_screens/chose_event_date_and_time.dart

```diff
@@ -20,30 +21,40 @@ class ChoseEventDateAndTime extends StatelessWidget {
         final command = state.pageCommand;
         BlocProvider.of<CreateRegionEventBloc>(context).add(const ClearCreateRegionEventPageCommand());
         if (command is ShowStartTimePicker || command is ShowEndTimePicker) {
-          final initialTime = command is ShowStartTimePicker ? DateTime.now() : state.eventStartTime ?? DateTime.now();
-          showTimePicker(context: context, initialTime: TimeOfDay.fromDateTime(initialTime)).then((selected) {
+          final initialTime = state.eventEndTime ?? TimeOfDay.now();
+          showTimePicker(context: context, initialTime: initialTime).then((selected) {
             if (selected != null) {
               BlocProvider.of<CreateRegionEventBloc>(context).add(
                 command is ShowStartTimePicker ? OnStartTimeChanged(selected) : OnEndTimeChanged(selected),
               );
             }
           });
-        } else if (command is ShowStartDatePicker || command is ShowEndDatePicker) {
-          final initialDate = command is ShowStartDatePicker ? DateTime.now() : state.eventStartTime ?? DateTime.now();
+        } else if (command is ShowStartDatePicker) {
+          final endDate = state.eventEndDate ?? DateTime(2099);
+          showDatePicker(
+            context: context,
+            initialDate: DateTime.now(),
+            firstDate: DateTime.now(),
+            lastDate: endDate,
+          ).then((selected) {
+            if (selected != null) {
+              BlocProvider.of<CreateRegionEventBloc>(context).add(OnStartDateChanged(selected));
```

Comment:
why not fire this event, 
let the bloc decide how to handle a null selected 
  if (selected != null) move to bloc